### ISSUE TITLE
Backup and Restore tasks show on Azure DB Manage page

### ISSUE THREAD

**Issue by kburtram** [Microsoft/sqlopsstudio#815]
- SQL Operations Studio Version: `master`

Steps to Reproduce:
1. Connect to an Azure SQL DB and open Database Manage page

Expected: No tasks that won't work are visible

Actual: Backup and Restore tasks buttons are visible

**Comment by kburtram** [comment]
@anthonydresser I still see the backup\restore tasks on the Azure DB Database page on the dashboard.

**Comment by kburtram** [comment]
Weird, the tasks are now gone from the dashboard, so I guess I'll close again pending further testing. 

I even captured a screenshot with the tasks on the dashboard.  Not sure why there're gone now.

<img width="744" alt="screen shot 2018-03-26 at 12 39 20 pm" src="https://user-images.githubusercontent.com/599935/37930050-6ffe5288-30f6-11e8-9f7b-1cb18d640d03.png">

**Comment by kburtram** [comment]
There's back again on the current test build from release branch.  There's something going on here.

**Comment by kburtram** [comment]
..the commands fail somewhat gracefully so I don't think this is a ship blocker, but certainly something to include in a hotfix build if we need to do that this month.

**Comment by kburtram** [comment]
The repro seems like the tasks only show the first time the dashboard is display.  If you navigate to the server or close and reopen the dashboard the backup\restore tasks go away.  If you start and stop SQL Ops they come back the first time the dashboard is opened.

**Comment by anthonydresser** [comment]
Do you have any user settings set for the dashboard?

**Comment by kburtram** [comment]
@anthonydresser yes, it appears I have the following dashboard settings.  Though I didn't add any of these settings.  Of course I can clear these settings, but if this is an issue that will impact many users upgrading then asking all of them to manually clean-up there settings seems inconvenient.

Any ideas what's going on and how broadly this will impact users? 

```
"dashboard.server.widgets": [
        {
            "gridItemConfig": {
                "sizex": 1,
                "sizey": 2,
                "col": 1,
                "row": 1
            },
            "widget": {
                "explorer-widget": {}
            }
        },
        {
            "widget": {
                "backup-history-server-insight": {
                    "cacheId": "0c7cba8b-c87a-4bcc-ae54-2f40a5503a90"
                }
            },
            "gridItemConfig": {
                "col": 2,
                "row": 1
            }
        },
        {
            "name": "Tasks",
            "widget": {
                "tasks-widget": {}
            },
            "gridItemConfig": {
                "sizex": 2,
                "sizey": 1,
                "col": 4,
                "row": 1
            }
        },
        {
            "widget": {
                "all-database-size-server-insight": {
                    "cacheId": "1d7cba8b-c87a-4bcc-ae54-2f40a5503a90"
                }
            },
            "gridItemConfig": {
                "col": 2,
                "row": 2,
                "sizex": 3,
                "sizey": 2
            }
        }
    ],
    "dashboard.server.tabs": [],
    "dashboard.database.tabs": [],
```

**Comment by anthonydresser** [comment]
`"tasks-widget": {}` this is probably whats causing it. We changed the enforcement for how stuff shows up to this declaration.

The default is `'tasks-widget': [{ name: 'backup', when: '!mssql:iscloud' }, { name: 'restore', when: '!mssql:iscloud' }, 'configureDashboard', 'newQuery']` since your's is blank, it will show all tasks registered regardless of anything.

**Comment by anthonydresser** [comment]
I'm not sure why its intermittent it should just always show everything based on your settings.
